applications/StructuralMechanicsApplication/python_scripts/python_solvers_wrapper_structural.py

```diff
@@ -8,6 +8,13 @@
 
 def CreateSolverByParameters(model, solver_settings, parallelism):
 
+    if solver_settings.Has("custom_solver_module"):
```

Comment:
my idea was even easier than this:

if one has a "solver_module" which is not in the wrapper list than it is treated as the name of a python file...

what do you think?

Reply:
hm this was also my first idea but this will require some refactoring in the solvers_wrapper due to how it is working now (many if etc)

If you want I can do that

Reply:
also we don't have any setting named "solver_module", only solver-type, which is not the same
Hence I made an extra one to make it very clear what is happening

Reply:
actually i am proposing to put such code in the wrapper itself. 

my point is that we could do programmatically the same change also in other applications in a very easy way.

Reply:
yes, but then you need to change all the solvers for this.

for the structural mechancis it is easy since they all derive from the same one. however in other applications it is not like this...

Reply:
> actually i am proposing to put such code in the wrapper itself.

I don't fully get you, could you please explain?

ofc if there would be a better/global solution I would like that :)

Reply:
no rocket science. i am proposing to change the solvers_wrapper as something like

~~~py
from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7

# Importing Kratos
import KratosMultiphysics

# Other imports
from importlib import import_module

def CreateSolverByParameters(model, solver_settings, parallelism):

    solver_type = solver_settings["solver_type"].GetString()

    if solver_settings.Has("time_integration_method"):
        time_integration_method = solver_settings["time_integration_method"].GetString()
    else:
        time_integration_method = "implicit" # defaulting to implicit time-integration

    # Solvers for OpenMP parallelism
    if parallelism == "OpenMP":
        if solver_type == "dynamic" or solver_type == "Dynamic":

        ...

        elif solver_type == "adjoint_static":
            solver_module_name = "structural_mechanics_adjoint_static_solver"

        else: #standard solver not found!
            try: #try treating "solver_type" as the name of the module
                solver = import_module(solvertype).CreateSolver(model, solver_settings)
                KratosMultiphysics.Logger.PrintInfo("MechanicalSolversWrapper", 'Using custom solver "{}", defined in module "{}"'.format(solver.__class__.__name__, solver.__class__.__module__))
                return solver
            except:
                err_msg =  "The requested solver type \"" + solver_type + "\" is not in the python solvers wrapper\n"
                err_msg += "Available options are: \"static\", \"dynamic\", \"eigen_value\", \"harmonic_analysis\", \"formfinding\", \"adjoint_static\""
                raise Exception(err_msg)

...
~~~

Reply:
ok now I understand

Reply:
I did the changes, but honestly I liked it better before :/

IMO the first version was cleaner and easier to read